src/app/core/storage.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  AngularFireStorage,
+  AngularFireStorageReference,
+  AngularFireUploadTask,
+} from '@angular/fire/storage';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StorageService {
+  public static imagesBaseFolder = 'messages';
+
+  constructor(private storageService: AngularFireStorage) {}
+
+  public static getFileExtension(file: File): string {
+    return file.name.split('.').pop();
+  }
+
+  private static generatePath(messageId: string, extension: string): string {
+    return `${this.imagesBaseFolder}/${messageId}/${uuidv4()}.${extension}`;
+  }
+
+  public async uploadImages(
+    messageId: string,
+    images: File[]
+  ): Promise<Promise<string>[]> {
```

Comment:
Do you really need to enclose all of this in a Promise? Couldn't you have this instead?
```suggestion
  public uploadImages(
    messageId: string,
    images: File[]
  ): Promise<string>[] {
```

Reply:
It could also be possible to return a `Promise<string[]>`, depending on what is the most useful here.

Reply:
@jonapap TypeScript doesn't allow it. I think it's because the function returns a promise, but so does the `.map()` function. I couldn't find a way to make `await` work on the `.map()` function, unfortunately.

Reply:
I'll try myself at it if I have time today. I mean, it's no big deal, but something feels wrong in returning a promise inside a promise

Reply:
I will also try using `Promise.all()` on it later today. That could work too since that's how I resolve the nested Promises when I call the function

Reply:
@jonapap see [36f7b685](https://github.com/virtrolio/virtrolio-site/pull/422/commits/36f7b685471c167789a953dfbe8414313b166fec)